tests/qchem/test_hamiltonians.py

```diff
@@ -107,6 +107,16 @@ def test_electron_integrals(symbols, geometry, core, active, e_core, one_ref, tw
     assert np.allclose(two, two_ref)
 
 
+def test_electron_integrals_openshell_warning():
+    r"""Test that electron_integrals raises an error when an open-shell molecule is provided."""
+    symbols = ["H", "H", "H"]
+    geometry = np.array([[0.0, 0.0, 0.0], [0.0, 0.0, 1.0], [0.0, 0.0, 2.0]], requires_grad=False)
+    mol = qchem.Molecule(symbols, geometry)
+
+    with pytest.raises(ValueError, match="Open-shell systems are not supported."):
+        qchem.electron_integrals(mol)()
+
```

Comment:
Only scf should be tested for this. Any reason why electron_integrals is checked? It calls scf anyways.